samcli/commands/local/lib/debug_context.py

```diff
@@ -13,6 +14,7 @@ def __init__(self,
         self.debug_port = debug_port
         self.debugger_path = debugger_path
         self.debug_args = debug_args
+        os.environ["PYTHONUNBUFFERED"] = "1" if self.debug_port else "0"
```

Comment:
Actually on second thought, can you skip the setting of "0" to facilitate usage of this class in multi-threaded scenarios. If you do 
```
if self.debug_port:
  os.environ[...] = 1
```

then you don't have the race condition to worry about.